great_tables/_locations.py

```diff
@@ -823,6 +832,45 @@ def resolve_rows_i(
     )
 
 
+def resolve_mask(
+    data: GTData | list[str],
+    expr: PlExpr,
+    excl_stub: bool = True,
+    excl_group: bool = True,
+) -> list[tuple[int, int, str]]:
+    """Return data for creating `CellPos`, based on expr"""
+    if not isinstance(expr, PlExpr):
+        raise ValueError("Only Polars expressions can be passed to the `mask` argument.")
+
+    frame: PlDataFrame = data._tbl_data
+    frame_cols = frame.columns
+
+    stub_var = data._boxhead.vars_from_type(ColInfoTypeEnum.stub)
+    group_var = data._boxhead.vars_from_type(ColInfoTypeEnum.row_group)
+    cols_excl = [*(stub_var if excl_stub else []), *(group_var if excl_group else [])]
+
+    # `df.select()` raises `ColumnNotFoundError` if columns are missing from the original DataFrame.
+    masked = frame.select(expr)
+    for col_excl in cols_excl:
+        masked = masked.drop(col_excl, strict=False)
+
+    # Validate that `masked.columns` exist in the `frame_cols`
+    if not (set(masked.columns).issubset(set(frame_cols))):
+        raise ValueError("The `mask` may reference columns not in the original DataFrame.")
```

Comment:
Could we enumerate the extra columns in the error message? For example, something like this?:

```python
missing = set(masked.columns) - set(frame_cols)

raise ValueError(
    "The `mask` expression produces extra columns, with names not in the original DataFrame."
    f"\n\nExtra columns: {missing}"
)
```